### PR TITLE
add postgres-helm-compat and modify postgres tests

### PR DESCRIPTION
**Pull request by joshrwolf**
adds a new `postgres-helm-compat` image intended to be compatible with the popular helm based postgresql installations (which is currently the bitnami helm chart).

This also removes our existing `postgres` helm chart based test, which is incompatible with the `postgres` image.

### CODE REVIEW

**Comment by found-it** [comment]
Should we add a `docker` based `tls` test for the regular postgres image?

**Comment by joshrwolf** [comment]
> Should we add a `docker` based `tls` test for the regular postgres image?

yes, but at the risk of giving a cop out. I think that test should belong at the package level instead of here

lmk if you disagree, if not I'll do a follow up in wolf/ios

**Comment by found-it** [comment]
I was thinking about this a bit today @joshrwolf . I'm still on the fence about whether the TLS test should land in the package itself. My one concern is about _when_ we run package tests. If we only run them at package build time, I don't think we would actually catch an issue like [this one](https://github.com/chainguard-images/images/issues/1898#issuecomment-1842092883). In that issue, the postgres package didn't change - it was the openssl version in the image that changed and broke the application.

Please correct me if I'm wrong but I don't think we have a way to re-run package tests based on changing dependencies. If we had a mechanism to re-run package tests when certain dependencies change then I think that will solve our issue since it includes both prebuilt images and adding the package on a wolfi-base. Otherwise right now I believe testing the image is really the only way we could catch an issue with dependencies drifting over time.

To sum up, at the risk of asking you to do even more work - my vote is still in the image. I think there's value in having a package test to make use of as we evolve package testing but I don't know if it would catch what we need it to right now.

**Comment by found-it** [comment]
Cool, yeah I like that for now so we can block updating the postgres image if there are TLS issues. Thanks for adding the test! Love the use of the k8s job btw

**Review by found-it — APPROVED** [review]
